experimenter/experimenter/experiments/tests/test_models.py

```diff
@@ -2351,6 +2351,92 @@ def test_get_branch_data_returns_correct_data(self):
         self.assertEqual(second["num_participants"], 75)
         self.assertEqual(second["description"], branch_b.description)
 
+    def test_get_metric_data_returns_correct_data(self):
+        experiment = NimbusExperimentFactory.create()
+        branch_a = NimbusBranchFactory.create(
+            experiment=experiment, name="Branch A", slug="branch-a"
+        )
+        branch_b = NimbusBranchFactory.create(
+            experiment=experiment, name="Branch B", slug="branch-b"
+        )
+
+        experiment.results_data = {
+            "v3": {
+                "overall": {
+                    "enrollments": {
+                        "all": {
+                            "branch-a": {
+                                "branch_data": {
+                                    "other_metrics": {
+                                        "retained": {
+                                            "absolute": {
+                                                "all": [{"lower": 1.49, "upper": 1.74}]
+                                            },
+                                            "relative_uplift": {
+                                                "control": {
```

Comment:
The key here represents which branch is being compared to the current branch (`"branch-a"`), so based on the branches in the experiment, this should be `"branch-b"`.
```suggestion
                                                "branch-b": {
```